File: src/test/resources/languages/ruby/valid.rb
class User
  attr_accessor :name, :email

  def initialize(name, email)
    @name = name
    @email = email
  end

  def to_s
    "#{@name} <#{@email}>"
  end
end

users = [
  User.new("Alice", "alice@example.com"),
  User.new("Bob", "bob@example.com")
]

users.each do |user|
  puts user
end

def greet(name)
  "Hello, #{name}!"
end

puts greet("Rubyist")

File: src/test/resources/languages/elixir/valid.ex
defmodule MyApp.User do
  defstruct [:id, :name, :email]

  def new(id, name, email) do
    %__MODULE__{id: id, name: name, email: email}
  end

  def greet(%__MODULE__{name: name}) do
    "Hello, #{name}!"
  end
end

defmodule MyApp.Main do
  def run do
    user = MyApp.User.new(1, "Elixir Dev", "dev@example.com")
    IO.puts MyApp.User.greet(user)
  end
end